javalib/src/main/scala/java/net/URI.scala

```diff
@@ -1,33 +1,1415 @@
+              return false
+            }
+            if (ipAddress.charAt(length - 1) != ']') {
+              return false
+            }
+            if ((ipAddress.charAt(1) == ':') && (ipAddress.charAt(2) != ':')) {
+              return false
+            }
+            offset = 1
+            if (length < 4) {
+              return false
+            }
+          case ']' =>
+            if (i != length - 1) {
+              return false
+            }
+            if (ipAddress.charAt(0) != '[') {
+              return false
+            }
+          case '.' => { numberOfPeriods += 1; numberOfPeriods - 1 }
+          if (numberOfPeriods > 3) {
+            return false
+          }
+          if (!isValidIP4Word(word)) {
+            return false
+          }
+          if (numberOfColons != 6 && !doubleColon) {
+            return false
+          }
+          if (numberOfColons == 7 && ipAddress.charAt(0 + offset) != ':' &&
+              ipAddress.charAt(1 + offset) != ':') {
+            return false
+          }
+          word = ""
+        case ':' => { numberOfColons += 1; numberOfColons - 1 }
+          if (numberOfColons > 7) {
+            return false
+          }
+          if (numberOfPeriods > 0) {
+            return false
+          }
+          if (prevChar == ':') {
+            if (doubleColon) {
+              return false
+            }
+            doubleColon = true
+          }
+          word = ""
+        case _ =>
+            if (word.length > 3) {
+              return false
+            }
+            if (!isValidHexChar(c)) {
+              return false
+            }
+            word += c
+
+        }
+      }
+      if (numberOfPeriods > 0) {
+        if (numberOfPeriods != 3 || !isValidIP4Word(word)) {
+          return false
+        }
+      } else {
+        if (numberOfColons != 7 && !doubleColon) {
+          return false
+        }
+        if (word == "" && ipAddress.charAt(length - 1 - offset) != ':' &&
+            ipAddress.charAt(length - 2 - offset) != ':') {
+          return false
+        }
+      }
+      true
+    }
+
+    def isValidIP4Word(word: String): Boolean = {
+      var c: Char = 0
+      if (word.length < 1 || word.length > 3) {
+        return false
+      }
+      for (i <- 0 until word.length) {
+        c = word.charAt(i)
+        if (!(c >= '0' && c <= '9')) {
+          return false
+        }
+      }
+      if (java.lang.Integer.parseInt(word) > 255) {
+        return false
+      }
+      true
+    }
+
+    def isValidHexChar(c: Char): Boolean =
+      (c >= '0' && c <= '9') || (c >= 'A' && c <= 'F') || (c >= 'a' && c <= 'f')
+
+  }
+
+  private def quoteComponent(component: String, legalset: String): String = {
+    try {
+      URIEncoderDecoder.quoteIllegal(component, legalset);
+    } catch {
+      case e: UnsupportedOperationException =>
+        throw new RuntimeException(e.toString)
+    }
+  }
+
+  def compareTo(uri: URI): Int = {
```

Comment:
Yup, when porting I forgot about the "return" difference in Java and Scala. I have corrected most of it, but I have missed some cases. I will fix them and double check.